apps/wallet/src/features/Swap/InputSwap/FromToken.tsx

```diff
@@ -0,0 +1,131 @@
+import { useMemo } from 'react';
+import { StyleSheet, TextInput } from 'react-native';
+import { Button, Text, View } from '@walless/gui';
+import { useSnapshot } from 'utils/hooks';
+
+import type { SwapContext } from '../context';
+import { swapActions, swapContext } from '../context';
+
+import SelectButton from './SelectButton';
+
+const FromToken = () => {
+	const { fromToken, amount } = useSnapshot(swapContext).swap as SwapContext;
+
+	const balance = useMemo(() => {
+		if (!fromToken) return 0;
+		return (
+			parseFloat(fromToken.account.balance) / 10 ** fromToken.account.decimals
+		);
+	}, [fromToken]);
+
+	const priceString = useMemo(() => {
+		if (!fromToken?.account.quotes?.usd) return '-';
+		if (!amount || isNaN(parseFloat(amount))) return '$0.00';
+		const price = parseFloat(amount) * fromToken.account.quotes?.usd;
+		return `$${price.toFixed(2)}`;
+	}, [amount]);
+
+	const handleSelectFromToken = () => {
+		swapActions.openSelectToken('from');
+	};
+
+	const updateAmount = (amount: string) => {
+		swapActions.update({ amount });
+	};
+
+	const handlePressMax = () => {
+		if (!fromToken) return;
+		const maxAmount =
+			parseFloat(fromToken.account.balance) / 10 ** fromToken.account.decimals;
```

Comment:
group `10 ** fromToken.account.decimals;` under variable or `()` to be more declarative.

Reply:
`()` is automatically removed by code linter